storm-core/src/clj/org/apache/storm/daemon/executor.clj

```diff
@@ -491,7 +493,7 @@
           EVENTLOGGER-STREAM-ID
           [component-id message-id (System/currentTimeMillis) values]))))
 
-(defmethod mk-threads :spout [executor-data task-datas initial-credentials]
+(defmethod mk-threads :spout [executor-data task-datas initial-credentials throttle-on]
```

Comment:
I think this is logically wrong. Throttle-on flag is a boolean that you have to obtain dynamically from the executor-data in run time, not that you can pass in as a constant when making the spout thread at the begining.